hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/HadoopConfigurations.java

```diff
@@ -63,6 +63,7 @@ public static org.apache.hadoop.conf.Configuration getHiveConf(Configuration con
     if (explicitDir != null) {
       hadoopConf.addResource(new Path(explicitDir, "hive-site.xml"));
     }
+    conf.toMap().forEach(hadoopConf::set);
     return hadoopConf;
```

Comment:
Seems you wanna pass around the hive config options through Flink sql options, this is not suggested, We prefer to config the hive properties through the `hive-site.xml` in the classpath.

Reply:
Not only the configuration of hive-site.xml, but also the storage configuration is also wanted to be passed in dynamically to the bottom layer because the two configurations are not separated.The goal is for the upper-layer application to be able to dynamically switch the catalog and switch the underlying storage.

Reply:
Does the `hadoop.` prefixed options work for your use case?

Reply:
If the storage is hdfs, hadoop prefix filtering is fine. If it is minio, connection parameters are filtered out.

Reply:
If the storage is hdfs, hadoop prefix filtering is fine. If it is minio, connection parameters are filtered out.

Reply:
Can we still prefix the minio params with the same prefix `hadoop.`

Reply:
If the storage is hdfs, hadoop prefix filtering is fine. If it is minio, connection parameters are filtered out.

Reply:
thanks,At present, it's not working

Reply:
Can you explain the reason, all the options prefixed with `hadoop.` are passed as hadoop configuration.

Reply:
thanks,the prefix hadoop. problem solved, but what about multi-catalog?

Reply:
What do you mean by muti-catalog, are there any catalog scope configurations?

Reply:
Users can create catalogs on the application side, and the storage may be different (hivemetastore can be shared). We will record these storage information and hive information, and then use it when he builds databases and tables, so it is multiple catalogs. So the storage parameters are passed from the catalog instead of getting them from the hadoop file of the environment configuration